tool-openssl/x509.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/x509.h>
+#include <openssl/pem.h>
+#include <stdio.h>
+#include <string>
+#include <vector>
+#include "../tool/internal.h"
+#include "internal.h"
+
+static const argument_t kArguments[] = {
+  { "-in", kRequiredArgument, "Input file" },
+  { "-out", kRequiredArgument, "Output file" },
+  { "", kOptionalArgument, "" }
+};
+
+// Map arguments using tool/args.cc
+bool X509Tool(const args_list_t &args) {
```

Comment:
Another thought that I had was about [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). For example, this `X509Tool` function is handling both the validation/parsing of arguments as well as performing the operation required for the tool. These are technically different "concerns". For this logic to more easily be unit-tested and to scale more efficiently, the argument parsing/validation should be separated from the operation it performs.

That said, "separation of concerns" requires more upfront effort and initially adds more structure/complexity to the code, specifically around the interfacing between the previously-unified-now-separated components. That additional effort is not always worth it early in a project or when the scaling will never be needed.